SPECS/tpm2-tss/tpm2-tss.spec

```diff
@@ -1,94 +1,75 @@
-Source0:        https://github.com/tpm2-software/tpm2-tss/releases/download/%{version}/%{name}-%{version}.tar.gz
-Patch0:         CVE-2023-22745.patch
+
+Source0: https://github.com/tpm2-software/tpm2-tss/releases/download/%{version}/%{name}-%{version}.tar.gz
+%define sha512 %{name}=cabb411f074dfa94919ba914849aac77a0ac2f50622e28a1406cf575369148774748e0e2b7a7c566ec83561a96d4b883bac5a3b1763f4cf48668a0c5d68c0a23
+Source1: %{name}.sysusers
 BuildRequires:  json-c-devel
 BuildRequires:  openssl-devel
-Requires:       json-c
-Requires:       openssl
-Requires(postun): %{_sbindir}/groupdel
-Requires(postun): %{_sbindir}/userdel
-Requires(pre):  %{_sbindir}/groupadd
-Requires(pre):  %{_sbindir}/useradd
+BuildRequires:  systemd-devel
+BuildRequires:  shadow-utils
+
+Requires: openssl
+Requires: systemd-rpm-macros
 
 %description
 OSS implementation of the TCG TPM2 Software Stack (TSS2)
 
-%package devel
-Summary:        The libraries and header files needed for TSS2 development.
-Requires:       %{name} = %{version}-%{release}
+%package      devel
+Summary:      The libraries and header files needed for TSS2 development.
+Requires:     %{name} = %{version}-%{release}
 
-%description devel
+%description    devel
 The libraries and header files needed for TSS2 development.
 
 %prep
 %autosetup -p1
 
 %build
 %configure \
-    --disable-static \
-    --disable-doxygen-doc \
-    --with-udevrulesdir=%{_sysconfdir}/udev/rules.d
+  --disable-static \
+  --disable-doxygen-doc \
+  --enable-fapi=no \
+  --with-udevrulesdir=%{_sysconfdir}/udev/rules.d
 
-make %{?_smp_mflags}
+%make_build
 
 %install
-make DESTDIR=%{buildroot} install
-find %{buildroot} -type f -name "*.la" -delete -print
+%make_install %{?_smp_mflags}
+install -p -D -m 0644 %{SOURCE1} %{buildroot}%{_sysusersdir}/%{name}.sysusers
 
 %post
+%sysusers_create_compat %{SOURCE1}
+mkdir -p /var/lib/tpm
+chown -R tss:tss /var/lib/tpm
 /sbin/ldconfig
-mkdir -p %{_sharedstatedir}/tpm
-if [ $1 -eq 1 ]; then
-    # this is initial installation
-    if ! getent group tss >/dev/null; then
-        groupadd tss
-    fi
-    if ! getent passwd tss >/dev/null; then
-        useradd -c "TCG Software Stack" -d %{_sharedstatedir}/tpm -g tss \
-            -s /bin/false tss
-    fi
-fi
-
-%postun
-/sbin/ldconfig
-if [ $1 -eq 0 ]; then
-    # this is delete operation
-    if getent passwd tss >/dev/null; then
-        userdel tss
-    fi
-    if getent group tss >/dev/null; then
-        groupdel tss
-    fi
-fi
 
 %files
 %defattr(-,root,root)
-%license LICENSE
 %{_sysconfdir}/udev/rules.d/tpm-udev.rules
-%{_sysconfdir}/tmpfiles.d/tpm2-tss-fapi.conf
-%{_sysconfdir}/tpm2-tss/*
-%{_libdir}/*.so.0.0.0
-%exclude %{_sysconfdir}/sysusers.d/tpm2-tss.conf
+%{_libdir}/*.so.*
+%{_sysusersdir}/%{name}.sysusers
 
 %files devel
 %defattr(-,root,root)
 %{_includedir}/tss2/*
 %{_libdir}/pkgconfig/*
 %{_libdir}/*.so
-%{_libdir}/*.so.0
-%{_mandir}/man3/*
-%{_mandir}/man5/*
-%{_mandir}/man7/*
+%{_libdir}/*.la
```

Comment:
why are you packaging the .la files?

Reply:
I would suggest doing something similar to what fedora does in the libtpms PR you have, e.g.: https://github.com/microsoft/CBL-Mariner/pull/7378/files#diff-6b1fd9397492e87896350f5cacbd559df43cba9ca7146ec0f92b308fd26c70e7R47

that looks like it's exactly what fedora does:
https://src.fedoraproject.org/rpms/tpm2-tss/blob/rawhide/f/tpm2-tss.spec#_75

I'm not sure why Photon doesn't need to remove the .la files but also doesn't package them.